src/strtree.c

```diff
@@ -38,17 +70,52 @@ static PyArrayObject *copy_kvec_to_npy(npy_intp_vec *arr)
     // in numpy.  There doesn't appear to be anything we can do to avoid it.
     PyArrayObject *result = (PyArrayObject *) PyArray_SimpleNew(1, dims, NPY_INTP);
     if (result == NULL) {
+        PyErr_SetString(PyExc_TypeError, "could not allocate numpy array");
         return NULL;
     }
 
     for (i = 0; i<size; i++) {
-        ptr = PyArray_GETPTR1(result, i);
-        *ptr = kv_A(*arr, i);
+        // assign value into numpy array
+        *(npy_intp *)PyArray_GETPTR1(result, i) = kv_A(*arr, i);
     }
 
     return (PyArrayObject *) result;
 }
 
+/* Copy values from arr and arr2 to a new 2D numpy integer array.
+ * arr and arr2 must be the same length.
+ */
+
+static PyArrayObject *copy_kvec_to_npy2(npy_intp_vec *arr, npy_intp_vec *arr2)
```

Comment:
We can't create a tuple of 2 ndarrays with the python C API to return? (to avoid this extra copy?)